### ISSUE TITLE
The MAP of auther's pre-trained model EfficientDet-d0 on VOC is about 0.56?

### ISSUE THREAD

**Issue by majian-stu** [toandaominh1997/EfficientDet.Pytorch#119]
I have run the script eval.py with the model EfficientDet-d0 offered by author, but the MAP is only about 0.56. Do you know why? I hope you can help me. Thank you!

**Comment by linerfei** [comment]
My results is 0.56 either, not 0.62.  By the way, I trained the network without pre trained network, and the results is lower than 0.36. SGD or adamW reaches  the same results , why?

**Comment by majian-stu** [comment]
> My results is 0.56 either, not 0.62.  By the way, I trained the network without pre trained network, and the results is lower than 0.36. SGD or adamW reaches  the same results , why?

Thank you for your reply. I also encountered this problem when training for both VOC and COCO dataset. Although the loss has been reduced to a very low level, mAP is poor.

**Comment by quangtn266** [comment]
> > My results is 0.56 either, not 0.62.  By the way, I trained the network without pre trained network, and the results is lower than 0.36. SGD or adamW reaches  the same results , why?
> 
> Thank you for your reply. I also encountered this problem when training for both VOC and COCO dataset. Although the loss has been reduced to a very low level, mAP is poor.

I got the same issue. After trying to fix this issue for 1 month, I decide to change another model :))

**Comment by majian-stu** [comment]
> My results is 0.56 either, not 0.62.  By the way, I trained the network without pre trained network, and the results is lower than 0.36. SGD or adamW reaches  the same results , why?
> 
> Thank you for your reply. I also encountered this problem when training for both VOC and COCO dataset. Although the loss has been reduced to a very low level, mAP is poor.
> 
> I got the same issue. After trying to fix this issue for 1 month, I decide to change another model :))

That‘s a sad story. I really hope the author can reply us...

**Comment by linerfei** [comment]
@majian-stu ,have you used pre trained efficientNet-B0 ? I changed the code 'from_pretrained('efficientnet-b0') ' and get from_name('efficientnet-b0') instead. I think this may lead to lower ap.

**Comment by majian-stu** [comment]
> @majian-stu ,have you used pre trained efficientNet-B0 ? I changed the code 'from_pretrained('efficientnet-b0') ' and get from_name('efficientnet-b0') instead. I think this may lead to lower ap.

Yes, I used the params from pre-trained model efficientnet-b0, Do you means I should use the params from efficientdet-d0 feeding into backnone?

**Comment by majian-stu** [comment]
> @majian-stu ,have you used pre trained efficientNet-B0 ? I changed the code 'from_pretrained('efficientnet-b0') ' and get from_name('efficientnet-b0') instead. I think this may lead to lower ap.
> 
> Yes, I used the params from pre-trained model efficientnet-b0, Do you means I should use the params from efficientdet-d0 feeding into backnone?

I have change the code "self.backbone = EfficientNet.from_pretrained(MODEL_MAP[network])" to "self.backbone = EfficientNet.from_name(MODEL_MAP[network])", and run eval.py again, but the mAP is still 0.56.

**Comment by kentaroy47** [comment]
the accuracy is quite strange.
our code provides mAP of 77%, but its still low. (should get ~85% with the paper's reported accuracy). Maybe adding image augumentations will get some accuracy boost?
I think lots of lr scheduling configurations will be required to get good performance on coco.
https://github.com/kentaroy47/efficientdet.pytorch

**Comment by y78h11b09** [comment]
our trained model of efficient-d0 provides mAP of 74.8% on VOC 07, and 70.7% on VOC12

**Comment by y78h11b09** [comment]
My model of efficient-d0 trained 255 epoches provides mAP of 74.8% on VOC 07, and 70.7% on VOC12

**Comment by xpngzhng** [comment]
@kentaroy47 hi, I have read your repo. Have you try coco dataset? 
I made some modifications to the bifpn implementation in this repo and add it to mmdetection, using  efficientdet-d2 I obtain mAP 0.347

**Comment by kentaroy47** [comment]
@xpngzhng 
I actually haven't finished coco training. That's a good coco accuracy even compared to faster-rcnns!

**Comment by djaym7** [comment]
> My model of efficient-d0 trained 255 epoches provides mAP of 74.8% on VOC 07, and 70.7% on VOC12

For testing on 2007 did you use 2007+2012 trainval for training the model ?
